.github/actions/deploy_docs/action.yml

```diff
@@ -0,0 +1,46 @@
+name: 'Deploy documentation'
+description: 'Deploy ESPResSo documentation'
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        # checkout github pages
+        cd "${HOME}"
+        git clone --quiet git@github.com:espressomd/espressomd.github.io.git
+        cd espressomd.github.io
+        LAST_COMMIT=$(git log -1 --pretty=format:"%s" remotes/origin/main -- doc dox tutorials)
+        NEXT_COMMIT="Documentation for ${GITHUB_SHA}"
+        if [ "${NEXT_COMMIT}" = "${LAST_COMMIT}" ]; then
+          echo "Documentation already up-to-date.";
+          exit 0;
+        fi
+        # download artifacts
+        curl https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/artifacts/python/download?job=run_tutorials -L -o tutorials.zip
+        curl https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/artifacts/python/download?job=run_doxygen   -L -o doxygen.zip
+        curl https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/artifacts/python/download?job=check_sphinx  -L -o sphinx.zip
+        if grep -F '<!DOCTYPE html>' *.zip; then
+          echo "The artifacts could not be downloaded.";
+          exit 1;
+        fi
+        unzip -q '*.zip'
+        # create a fresh main branch
+        git config --global user.email "noreply@icp.uni-stuttgart.de"
+        git config --global user.name "espresso-ci"
+        git checkout -b new_main remotes/origin/releases
+        # generate the landing page
```

Comment:
Please add a comment mentioning the branch structure of the github pages repo and how the final page is constructed.